doc/autoyast.md

```diff
@@ -1,141 +1,101 @@
-check [proposal.md](./proposal.md).
+* [General settings][1], which basically allows to alter the `partitioning` section of the
+  product features, although only two options are documented (and thus supported).
+* [Partitioning layout][2], which defines how the partitioning should look like (partitions, volume
+  groups, etc.).
 
+Translating this to use cases levels:
 
-## Level 3 of AutoYaST and the installation proposal
+* Level 1: no settings at all.
+* Level 2: just _general settings_.
+* Level 3: _partitioning layout settings_ and, optionally, _general settings_.
 
-[This gist](https://gist.github.com/imobachgs/0f4049c2955b858c0713896210306aa1)
-presents many use cases of the so-called level 3 of AutoYaST partitioning, based
-on examples found in the AutoYaST documentation.
+## Process Overview
 
-The process followed by AutoYaST is quite different from the proposal one, but
-the concept of `PlannedPartition`, `PlannedLv` and several auxiliary classes
-that take list of those as input to perform operations can be reused to a big
-extent.
+The AutoYaST partitioning process starts in the [storage_auto client][3], that
+lives in the `autoyast2-installation` package. This client relies heavily on the
+[AutoinstStorage module][4] which is responsible, among other things, of
+importing the AutoYaST profile (via the `#Import` method) and performing the
+partitioning (`#Write`).
 
+This document focuses in the `#Import` method, which creates a partitioning
+proposal and provides libstorage-ng with the desired *devicegraph*. The
+following is just an sketch of the steps of the partitioning process (the rest
+of the sections will dig into the details for all of them):
 
-## Adapting the existing code
+1. The general settings are [imported][5]. It happens before the
+   `AutoinstStorage#Import` method is actually called.
+2. The settings are [preprocessed][6]. Currently, the only preprocessing which
+   takes place is replacing `device=ask` with a real device name after asking
+   the user. Any preprocessing which implies interacting with the user should be
+   done at this point.
+3. Build the proposed [devicegraph][7]. This is the complex part and it involves a
+   lot of stuff that will be described later.
+4. If [any issue][8] is found during step 3, [present][9] it to the user.
 
-### Features that can be added to the existing components/classes
+The third step is quite generic and it is completely different depending on
+whether a `partitioning` section is present in the profile or not. The
+[Y2Autoinstall::StorageProposal][10] class decides which approach to follow.
 
-The following features could, at first sight, be added to the current system
-with a relatively low impact in the current structure and philosophy of the
-code.
+If there is no `partitioning` section, AutoYaST simply relies in the
+{Y2Storage::GuidedProposal}. Bear in mind that such a proposal can be influenced
```

Comment:
What's the intention of these braces? I don't believe they mean anything in markdown

Reply:
Nevermind, I guess they work just fine when the file is rendered via yardoc.

Which made me notice this looks outdated https://www.rubydoc.info/github/yast/yast-storage-ng/master/file/

Reply:
Never mind again. Now it looks up-to-date :-)